src/Polymorph-Widgets/UIThemePalette.class.st

```diff
@@ -0,0 +1,659 @@
+Class {
+	#name : #UIThemePalette,
+	#superclass : #Object,
+	#instVars : [
+		'colorConfigurator',
+		'paletteDictionary'
+	],
+	#category : #'Polymorph-Widgets-Themes'
+}
+
+{ #category : #colors }
+UIThemePalette >> backgroundColor [
+
+	^ paletteDictionary at: #backgroundColor ifAbsentPut: [ colorConfigurator backgroundColor ]
+]
```

Comment:
These methods make me think if we don't want (or it's already implemented maybe) some mechanism to cache the result of a method.

I'm thinking something like
```st
UIThemePalette >> backgroundColor [
	<cached>

	^ colorConfigurator backgroundColor
]
```
And the magic happens (the first time the method is executed, the sentence is evaluated and the returned object is stored, then it returns the stored result for the next invocations).

Reply:
And how is it invalidated? 🤔

Reply:
there is already `[] memoised`. But, how is it invalidated ? :)

Reply:
When you change the theme, for example from light to dark, you create a new palette object with a new color configurator. That is how is invalidated. Nevertheless, there is no an api to invalidate a color without changing the theme, this is no different from current's implementation

Reply:
I like the <cached> pragme that Nahuel proposed. But for this is a separated project. Maybe a Phep?